dbg.h

```diff
@@ -202,13 +215,13 @@ using std::size;
 }  // namespace
 
 template <typename T>
-using detect_begin_t = decltype(begin(std::declval<T>()));
+using detect_begin_t = decltype(detail::begin(std::declval<T>()));
```

Comment:
These `detail::` namespace annotations fix a compiler error on MSVC and seem to work fine on GCC and Clang. However, to be honest, I have never completely understood ADL, so I can't really tell if that causes any other issues.

CC @hmenke: only if you are interested, of course.

Reply:
Using ADL was probably a bad idea anyway.